packages/server-renderer/src/render.ts

```diff
@@ -188,7 +188,7 @@ function renderComponentSubTree(
       const prev = setCurrentRenderingInstance(instance)
       try {
         ssrRender(
-          instance.proxy,
+          getExposeProxy(instance) || instance.proxy,
```

Comment:
This isn't really the right fix as for the component's own render function, it should be using the default (internal facing) proxy.

See the correct fix in df686abb4 (reusing tests from this PR)